src/MomentoCacheClient.php

```diff
@@ -1443,11 +1459,67 @@ public function select(int $db): Redis|bool
         throw MomentoToPhpRedisExceptionMapper::createCommandNotImplementedException(__FUNCTION__);
     }
 
+    /**
+     * @throws Exception
+     */
     public function set(string $key, mixed $value, mixed $options = null): Redis|string|bool
     {
-        $result = $this->client->set($this->cacheName, $key, $value, $options);
+        if (is_numeric($value)) {
+            $value = (string)$value;
+        }
+
+        $ttl = null;
+
+        if (is_array($options)) {
+            $options = array_change_key_case($options, CASE_LOWER);
+
+            // Handle different TTL options (EX, PX, EXAT, PXAT)
+            if (isset($options['ex'])) {
+                $ttl = $options['ex'];
+            } elseif (isset($options['px'])) {
+                $ttl = $options['px'] / 1000;
+            } elseif (isset($options['exat'])) {
+                $ttl = $options['exat'] - time();
+            } elseif (isset($options['pxat'])) {
+                $ttl = floor(($options['pxat'] - microtime(true) * 1000) / 1000);
+            } elseif (isset($options['keepttl'])) {
+                throw MomentoToPhpRedisExceptionMapper::createArgumentNotSupportedException('set', 'keepttl');
+            }
+
+            // Handle NX option: Set if the key does not exist
+            if (in_array('nx', $options, true)) {
+                $result = $this->client->setIfAbsent($this->cacheName, $key, $value, $ttl);
+                if ($result->asStored()) {
+                    return "OK";
+                } else if ($result->asNotStored()) {
+                    return false;
+                } else {
+                    return MomentoToPhpRedisExceptionMapper::mapException($result);
+                }
+            }
+
+            // Handle XX option: Set only if the key already exists
+            if (in_array('xx', $options, true)) {
```

Comment:
Should make the xx case `else if`. The phpredis client may have some priority order for checking if multiple are passed in. This is unlikely but we should only perform one action in that event.